README.md

```diff
@@ -15,7 +15,7 @@ engines:
     config:
       patterns:
         no-set-methods:
-          pattern: /def set_\w+/
+          pattern: "def set_\\w+"
```

Comment:
FWIW the YAML behavior may depend a bit on quoting.

e.g.

```
engines:
  grep:
    enabled: true
    config:
      thing: foo\.bar
```

```
[1] pry(main)> c = CC::Yaml.parse(File.read("foo.yml"))
=> #<CC::Yaml::Nodes::Root:{"engines"=>{"grep"=>{"enabled"=>true, "config"=>{"thing"=>"foo\\.bar"}}}}>
[3] pry(main)> c.engines["grep"].config
=> {"thing"=>"foo\\.bar"}
# the extra `\` is ruby inserting it for display purposes: it's not really there
[10] pry(main)> puts c.engines["grep"].config["thing"][2]
o
[8] pry(main)> puts c.engines["grep"].config["thing"][3]
\
=> nil
[9] pry(main)> puts c.engines["grep"].config["thing"][4]
.
=> nil
```

But:

```
engines:
  grep:
    enabled: true
    config:
      thing: "foo\.bar"

```

```
[11] pry(main)> c = CC::Yaml.parse(File.read("foo.yml"))
=> #<CC::Yaml::Nodes::Root:{}>
[12] pry(main)> c.errors
=> ["syntax error: (<unknown>): found unknown escape character while parsing a quoted scalar at line 5 c
```

When explicitly quoting the expression, double-escaping it as `"foo\\.bar"` goes back to working correctly. Double-escaping when *not* quoting does result in two actual `\` in the string after parsing.

Saw pretty much the same behavior in all cases with plain `Psych` as well. So YAML is certainly wack, but at least our wrapper doesn't seem to be adding much wackness in this case.